### DISCUSSION TITLE
Getting CEK

### DISCUSSION THREAD

**Discussion by avator888** [lestrrat-go/jwx#1001] [Q&A]
Hello All,
Thank you for the project. It seems very hard to find an implementation of JWX in go world. 
I came across an issue that seems to be easy at first; however, I was not able to find solution yet. 

What I am trying to do:
1. building JWE message to send to the server and sending JWE to the server
2. server get the message and reply with data that encrypted using my CEK
3.  getting message from sever and decrypting it using CEK that was generated on step #1 

step 1 could be implemented using example:
https://github.com/lestrrat-go/jwx/blob/develop/v2/examples/jwe_example_test.go
> encrypted, err := jwe.Encrypt(payload, jwe.WithKey(jwa.RSA1_5, &privkey.PublicKey), jwe.WithContentEncryption(jwa.A128CBC_HS256))

Unfortunately, I could not find a way to retrieve CEK from jwe.WithContentEncryption. Please ping me in right direction if I am missing something. Really hope if process of getting CEK somehow implemented. 

Steps 1 and 2 were described in example from RFC 7516
https://datatracker.ietf.org/doc/html/rfc7516#appendix-A

Appreciate any help possible 
=^_^=

**Comment by lestrrat** [comment]
I'm not sure what the question is. As an end user you should not need to directly interact with CEK. Did you try `jwe.Decrypt`?

**Comment by avator888** [comment]
Thank you for your answer. Really appreciate that. 
It could be cases when it is necessary to directly interact with CEK. For example, testing or adapting business logic that sometimes required re-use so to day of CEK. Since go does not have other lib that covers JWx standard I have to refer to example from java world
 https://connect2id.com/products/nimbus-jose-jwt/examples/jwe-with-preset-cek
Basically, CEK is set up and then passed to encryptor during encryption.

**Reply by lestrrat** [reply]
Oh I see, then there's no way to currently do that. I suppose there could be an API like this?

```go
var cek []byte
_, err := jwe.Decrypt(...., jwe.WithCEK(&cek))
if err != nil {
   ....
}

// use cek
```

**Reply by lestrrat** [reply]
@avator888 I whipped #1011 without much testing, but is this what you want?

(please see [test case](https://github.com/lestrrat-go/jwx/pull/1011/files#diff-1391fc81668eac5abfcc63d0243f88508a3f07b43c7fbbb7b4e9fde54fd792f8) for example usage)

**Reply by lestrrat** [reply]
FYI, the above has been merged, and will be released in a future version
